learning_track.md

```diff
@@ -0,0 +1,141 @@
+
+### Learn how to deploy the Elixir app on staging (aka "prochainement")
+
+* Use a force push of your branch, e.g. `git push <remote> <branche>:prochainement -f` (so if your branch is `some-feature`, this will usually be: `git push origin some-feature:prochainement -f`)
+* This will trigger a redeploy
+* If you see errors in the CC app logs due to Ecto migrations (due to divergence of branches), you'll want to reset the staging database (see below)
+
+### Learn how to reset the staging (aka "prochainement") database
+
+* Go to the CleverCloud dashboard for the production Postgres database and download it locally
+* Read the [restore_db.sh](https://github.com/etalab/transport-site/blob/master/restore_db.sh) script
+* Go to the CleverCloud dashboard for the **staging** Postgres database, and run `restore_db.sh` with proper parameters
+
+### Learn how to connect via SSH
+
+* Make sure to link the correct app (production or staging) with `clever link $$REPLACE_BY_APP_ID$$` (as displayed in the staging/production app CC dashboards)
+* Verify the linking status with `clever applications`
+* Log with the app alias: `clever ssh --alias transport-prochainement`
+
+### Explore the GBFS conversion proxy
+
+* Check out the code under `apps/gbfs`
+* Run the app locally with `mix phx.server`
+* Discover the available converted feeds via `http://localhost:5000/gbfs`
+* Note that this does not contain all the GBFS feeds, only the ones we convert
+
+### Learn about the GTFS and GTFS-RT specifications
+
+* @thbar bought https://gumroad.com/l/gtfsbundle (available to the team on demand)
+* Download a tiny GTFS ([example](https://transport.data.gouv.fr/datasets/horaires-theoriques-et-temps-reel-des-navettes-de-la-station-de-tignes-gtfs-gtfs-rt/))
+* Check out this [diagram](https://www.researchgate.net/figure/Modele-de-donnees-GTFS_fig7_268333353)
+* Read the GTFS guide
+* Read the GTFS-realtime guide
+
+### Explore the GTFS validator locally
+
+* Clone the [transport-validator](https://github.com/etalab/transport-validator) project locally
+* Install Rust
+* Install [Rust Analyzer](https://github.com/rust-analyzer/rust-analyzer) for VSCode completion
+* Run all the tests with `cargo run test`
+
+### Run the GTFS validator via the Elixir app locally
+
+* Compile the validator project with `cargo build --release`
+* Run it as a server with `./target/release/main`
+* Use the displayed host & port (e.g. `http://127.0.0.1:7878`) to fill your `.envrc` configuration (`export GTFS_VALIDATOR_URL=http://127.0.0.1:7878`)
+* Run the site with `mix phx.server`
+* Go to `http://localhost:5000/validate`
+* Upload a GTFS file
+* Verify that it goes through the Elixir apps logs, then the validator logs
+
+### Debug the GTFS validator locally
+
+* Make sure to have Rust Analyzer installed
+* Install the [CodeLLDB extension](https://marketplace.visualstudio.com/items?itemName=vadimcn.vscode-lldb) to help with debugging
```

Comment:
C'est un peu spécifique VSCode...peut-être juste rajouter `if you use VSCode`, mais c'est un détail.

Reply:
Bonne remarque, j'ai corrigé et ajouté des liens